Added the code with implementation of AES in CBC mode

Northstrix opened the pull request:
AES_in_CBC_mode_for_microcontrollers-master.zip contains the code that enables the developer to easily integrate the AES-128, AES-192, and AES-256 in CBC mode to the RTL8720DN project.

*By completing this PR sufficiently, please help us to review and improve the quality of this PR and Release Notes*

### Checklist
1. [x] Please provide specific title that describe the change (*eg. Update Readme.md*)
2. [x] Please provide related links (*eg. This Pull Request is a fix of #issue_number*)
3. [x] Please **update the license/declaration** if applicable (*eg. Usage of other projects/apis/libs code*)
4. [x] Please check [Contributing guide](https://github.com/ambiot/ambd_arduino/blob/dev/CONTRIBUTING.md)
5. [x] Please **confirm option to "Allow edits and access to secrets by maintainers"** when opening a Pull Request

*This entire section above can be deleted if all items are checked.*

-----------
## Description of Change
Please describe your proposed Pull Request and it's impact. \
*eg.*
- Feature:
- API Updates:
- Misc:

## Tests and Environments 
Please describe on what Hardware, SDK, IDE, OS that tested this Pull Request and how if necessary.  \
*eg.*
- AMB21
- ambd_arduino V3.1.7
- Arduino IDE 2.3.0
- Windows 10

## Remark related links
Please provide links to related issue, PRs etc. \
*eg.*
- Closes #issue_number

designer2k2 commented:
There is a huge image in this zip, 14.208 x 3124 pixels, with 2,37MB.  Is this really needed?

For example, just running the file over https://tinypng.com/ reduces it down to 0,3MB.

Also are you shure this is a valid arduino library?

Northstrix commented:
> There is a huge image in this zip, 14.208 x 3124 pixels, with 2,37MB. Is this really needed?
> 
> For example, just running the file over https://tinypng.com/ reduces it down to 0,3MB.
>

I updated the pull request and compressed the image.
Yes, Image is needed, it demonstrates how the algorithm works. The reason why I put it in the first place is that CBC mode can be implemented with or without the IV encryption. In that particular library IV is encrypted. Additionally I added the key incrementation after each block.


> Also are you shure this is a valid arduino library?

Yes, I'm sure. I've used this library in several of my projects. In the archive that I included to the pull request, I put the library and the code that enables the developer to utilize that library to encrypt and decrypt data in AES in CBC mode.

M-ichae-l commented:
Many thanks for the contribution.
